cpp/arcticdb/storage/library_manager.cpp

```diff
@@ -115,14 +119,61 @@ void LibraryManager::write_library_config(const py::object& lib_cfg, const Libra
     );
 }
 
+void LibraryManager::modify_library_option(const arcticdb::storage::LibraryPath &path,
+                                           std::variant<ModifiableLibraryOption, ModifiableEnterpriseLibraryOption> option,
+                                           std::variant<bool, int64_t> new_value) const {
+    // We don't apply a storage override to keep modification backwards compatible.
+    // Before v3.0.0 we didn't use storage overrides and applying one when modifying options would break readers older
+    // than v3.0.0.
+    arcticdb::proto::storage::LibraryConfig config = get_config_internal(path, std::nullopt);
+    auto mutable_write_options = config.mutable_lib_desc()->mutable_version()->mutable_write_options();
+
+    auto get_bool = [&option](std::variant<bool, int64_t> value){
+        util::check(std::holds_alternative<bool>(value), "{} only supports bool values but received {}. Not changing library option.", option, value);
+        return std::get<bool>(value);
+    };
+    auto get_positive_int = [&option](std::variant<bool, int64_t> value){
+        util::check(std::holds_alternative<int64_t>(value) && std::get<int64_t>(value)>0, "{} only supports positive int values but received {}. Not changing library option.", option, value);
+        return std::get<int64_t>(value);
+    };
+
+    util::variant_match(
+            option,
+            [&](const ModifiableLibraryOption& option){
+                switch (option) {
+                    case ModifiableLibraryOption::DEDUP:
+                        mutable_write_options->set_de_duplication(get_bool(new_value));
+                        break;
+                    case ModifiableLibraryOption::ROWS_PER_SEGMENT:
+                        mutable_write_options->set_segment_row_size(get_positive_int(new_value));
+                        break;
+                    case ModifiableLibraryOption::COLUMNS_PER_SEGMENT:
+                        mutable_write_options->set_column_group_size(get_positive_int(new_value));
+                        break;
+                }
+            }, [&](const ModifiableEnterpriseLibraryOption& option){
+                switch (option) {
+                    case ModifiableEnterpriseLibraryOption::REPLICATION:
+                        mutable_write_options->mutable_sync_passive()->set_enabled(get_bool(new_value));
+                        break;
+                    case ModifiableEnterpriseLibraryOption::BACKGROUND_DELETION:
+                        mutable_write_options->set_delayed_deletes(get_bool(new_value));
+                        break;
+                }
+            });
+
+    // We don't want to validate old pre v3.0.0 configs
```

Comment:
Comment is a bit confusing, mention that it's documenting the `false` parameter?

Reply:
Done